apis/duck/v1alpha1/condition_set.go

```diff
@@ -126,14 +118,23 @@ var _ ConditionManager = (*conditionsImpl)(nil)
 // +k8s:deepcopy-gen=false
 type conditionsImpl struct {
 	ConditionSet
-	accessor ConditionsAccessor
+	status interface{}
 }
 
-// Manage creates a ConditionManager from an object that implements
-// ConditionsAccessopr using the original ConditionSet as a reference.
-func (r ConditionSet) Manage(accessor ConditionsAccessor) ConditionManager {
+// Manage creates a ConditionManager from a status object using the original
+// ConditionSet as a reference. Status must be or point to a struct.
+func (r ConditionSet) Manage(status interface{}) ConditionManager {
+	accessorValue := reflect.Indirect(reflect.ValueOf(status))
```

Comment:
Why not store `accessorValue` in the return value? Do we need to deref the pointer each time if it is one? If so, please include a test which would fail if we stored `accessorValue`.

Reply:
interesting idea. I can fold that in. Thanks!